security/pkg/server/ca/server.go

```diff
@@ -271,7 +271,11 @@ func (s *Server) applyServerCertificate() (*tls.Certificate, error) {
 func (s *Server) authenticate(ctx context.Context) *authenticate.Caller {
 	// TODO: apply different authenticators in specific order / according to configuration.
 	for _, authn := range s.authenticators {
-		if u, err := authn.Authenticate(ctx); u != nil && err == nil {
+		u, err := authn.Authenticate(ctx)
+		if err != nil {
```

Comment:
Why not use else following the if?  It can still print <nil> if err is nil.

Reply:
using else works in current code, check u!=nil also does no harm(also safer:)